wd_core.au3

```diff
@@ -218,32 +218,31 @@ Global $_WD_SupportedBrowsers[][$_WD_BROWSER__COUNTER] = _
 ; ===============================================================================================================================
 Func _WD_CreateSession($sCapabilities = Default)
 	Local Const $sFuncName = "_WD_CreateSession"
-	Local $sSession = ""
+	Local $sSession = "", $sMessage = ''
 
 	If $sCapabilities = Default Then $sCapabilities = $_WD_EmptyDict
 
 	Local $sResponse = __WD_Post($_WD_BASE_URL & ":" & $_WD_PORT & "/session", $sCapabilities)
 	Local $iErr = @error
 
-	__WD_ConsoleWrite($sFuncName & ": " & $sResponse, $_WD_DEBUG_Info)
-
 	If $iErr = $_WD_ERROR_Success Then
 		Local $oJSON = Json_Decode($sResponse)
 		$sSession = Json_Get($oJSON, "[value][sessionId]")
 
 		If @error Then
-			Local $sMessage = Json_Get($oJSON, "[value][message]")
-
-			Return SetError(__WD_Error($sFuncName, $_WD_ERROR_Exception, $sMessage), 0, "")
+			$sMessage = Json_Get($oJSON, "[value][message]")
+			$iErr = $_WD_ERROR_Exception
 		EndIf
+
+		$sMessage = $sSession
+
+		; Save response details for future use
+		$_WD_SESSION_DETAILS = $sResponse
```

Comment:
Why is not saved only on sucess ?
What happend if I open one session (with success), close them, and finally will try to open new one (with error)?

Reply:
1) Because this variable is designed to hold the details from the last successful session creation
2) Script designer has no reason to access `$_WD_SESSION_DETAILS` if session creation failed
3) This UDF doesn't currently handle multiple sessions well (see #63)
4) This behavior hasn't changed in this PR. Suggest that any further discussion be moved to an issue.

Reply:
> 4\. This behavior hasn't changed in this PR. Suggest that any further discussion be moved to an issue.

in this new version it is set when
```autoit
If $iErr = $_WD_ERROR_Success Then
  $_WD_SESSION_DETAILS = $sResponse
Else
```
but previously it was:
```autoit
If $iErr = $_WD_ERROR_Success Then
Else
  $_WD_SESSION_DETAILS = $sResponse
EndIf
````

Reply:
But as I stated at first I open one session and then close them, after which I open second session and only second opening gives error.

I this scenario `$_WD_SESSION_DETAILS`  should be at least reset to empty string, and not staying with responsetext from first opened session.

Reply:
> but previously it was:
> If $iErr = $_WD_ERROR_Success Then
> Else
>   $_WD_SESSION_DETAILS = $sResponse
> EndIf

That is not correct. In the current release (v0.9.0) --

```
	If $iErr = $_WD_ERROR_Success Then
		Local $oJSON = Json_Decode($sResponse)
		$sSession = Json_Get($oJSON, "[value][sessionId]")

		If @error Then
			Local $sMessage = Json_Get($oJSON, "[value][message]")

			Return SetError(__WD_Error($sFuncName, $_WD_ERROR_Exception, $sMessage), 0, "")
		EndIf
	Else
		Return SetError(__WD_Error($sFuncName, $_WD_ERROR_Exception), 0, "")
	EndIf

	; Save response details for future use
	$_WD_SESSION_DETAILS = $sResponse
```

So the function was always exiting before `save response` under error conditions. Now it does this --

```
	If $iErr = $_WD_ERROR_Success Then
		Local $oJSON = Json_Decode($sResponse)
		$sSession = Json_Get($oJSON, "[value][sessionId]")

		If @error Then
			$sMessage = Json_Get($oJSON, "[value][message]")
			$iErr = $_WD_ERROR_Exception
		Else
			$sMessage = $sSession

			; Save response details for future use
			$_WD_SESSION_DETAILS = $sResponse			
		EndIf
	Else
		$iErr = $_WD_ERROR_Exception
	EndIf
```

I eliminated the multiple exit points, but the functionality should remain where `$_WD_SESSION_DETAILS` is only updated upon successful session creation.

Reply:
> I do not have 2 sessions at the same time, but two separate sessions in succession, one after the other, when the former is already closed, but the program is still running.

It really shouldn't matter as the script can't continue due to error in _WD_CreateSession and therefore there isn't any reason to be using the contents of `$_WD_SESSION_DETAILS`

> I this scenario $_WD_SESSION_DETAILS should be at least reset to empty string, and not staying with responsetext from first opened session.

While I don't disagree with this, there may be instances where making this change will negatively affect existing scripts.

Reply:
Ok.
Treat this part of coversation as resolved.